### ISSUE TITLE
[no-empty] Allow empty catch blocks

### ISSUE THREAD

**Issue by tscpp** [eslint/eslint#13889]
**What rule do you want to change?**
`no-empty`

**Does this change cause the rule to produce more or fewer warnings?**
It will cause fewer warnings if used empty catch blocks.

**How will the change be implemented? (New option, new default behavior, etc.)?**
Option or new default behavior

**Please provide some example code that this change will affect:**

```js
try {
    ...
} catch { }
```

**What does the rule currently do for this code?**
![image](https://user-images.githubusercontent.com/47453366/100519943-6f2f6100-319b-11eb-9447-a414701746bf.png)
```
try {
    ...
} catch { }
        ~~~
-------------------------------------------------
Empty block statement.
```

**What will the rule do after it's changed?**
Allow empty catch blocks in try statements

**Are you willing to submit a pull request to implement this change?**
no

**Comment by mdjermanovic** [comment]
Hi @tscpp, thanks for the issue!

I think this option already exists: [allowEmptyCatch](https://eslint.org/docs/rules/no-empty#allowemptycatch)

```js
/* eslint no-empty: ["error", { allowEmptyCatch: true }] */

try {
  foo();
} catch {}
```

[Online Demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyogZXNsaW50IG5vLWVtcHR5OiBbXCJlcnJvclwiLCB7IGFsbG93RW1wdHlDYXRjaDogdHJ1ZSB9XSAqL1xuXG50cnkge1xuICBmb28oKTtcbn0gY2F0Y2gge30iLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOjEwLCJzb3VyY2VUeXBlIjoic2NyaXB0IiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7fSwiZW52Ijp7fX19)
